syntax = "proto3";
package im.codec;

// import "/home/myself/workspace/MyChat/common/proto/base.proto";
import "base.proto";



// CodecService 提供编解码功能，允许其他语言通过gRPC调用
service CodecService {
  // 编码服务：将IMHeader和Protobuf消息编码为二进制数据
  rpc Encode(EncodeRequest) returns (EncodeResponse) {}
  
  // 解码服务：将二进制数据解码为IMHeader和Protobuf消息
  rpc Decode(DecodeRequest) returns (DecodeResponse) {}
}

// 编码请求
message EncodeRequest {
  im.base.IMHeader header = 1;     // 消息头
  bytes message_data = 2;          // 消息体的序列化数据
  string message_type_name = 3;    // 消息类型名称
}

// 编码响应
message EncodeResponse {
  bool success = 1;                // 是否编码成功
  string error_message = 2;        // 错误信息
  bytes encoded_data = 3;          // 编码后的二进制数据
}

// 解码请求
message DecodeRequest {
  bytes encoded_data = 1;          // 待解码的二进制数据
}

// 解码响应
message DecodeResponse {
  bool success = 1;                // 是否解码成功
  string error_message = 2;        // 错误信息
  im.base.IMHeader header = 3;     // 解析出的消息头
  bytes message_data = 4;          // 解析出的消息体数据
  string message_type_name = 5;    // 消息类型名称
}